fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/fast/ResourcePayloadReader.java

```diff
@@ -0,0 +1,683 @@
+
+    // The resource type class of the resources being exported by this instance (derived from the injected fhirResourceType value
+    Class<? extends Resource> resourceType;
+
+    /**
+     * Fhir tenant id.
+     */
+    @Inject
+    @BatchProperty(name = Constants.FHIR_TENANT)
+    String fhirTenant;
+
+    /**
+     * Fhir data store id.
+     */
+    @Inject
+    @BatchProperty(name = Constants.FHIR_DATASTORE_ID)
+    String fhirDatastoreId;
+
+    /**
+     * Fhir resource type to process.
+     */
+    @Inject
+    @BatchProperty(name = Constants.PARTITION_RESOURCE_TYPE)
+    String fhirResourceType;
+
+
+    // TODO ???
+    private boolean isExportPublic = true;
+
+    // The IBM COS API key or S3 access key.
+    @Inject
+    @BatchProperty(name = Constants.COS_API_KEY)
+    String cosApiKeyProperty;
+
+    /**
+     * The IBM COS service instance id or s3 secret key.
+     */
+    @Inject
+    @BatchProperty(name = Constants.COS_SRVINST_ID)
+    String cosSrvinstId;
+
+    /**
+     * The Cos End point URL.
+     */
+    @Inject
+    @BatchProperty(name = Constants.COS_ENDPOINT_URL)
+    String cosEndpointUrl;
+
+    /**
+     * The Cos End point location.
+     */
+    @Inject
+    @BatchProperty(name = Constants.COS_LOCATION)
+    String cosLocation;
+
+    /**
+     * The Cos bucket name.
+     */
+    @Inject
+    @BatchProperty(name = Constants.COS_BUCKET_NAME)
+    String cosBucketName;
+
+    /**
+     * The Cos bucket path prefix.
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_COS_OBJECT_PATHPREFIX)
+    String cosBucketPathPrefix;
+
+    /**
+     * If use IBM credential or Amazon secret keys.
+     */
+    @Inject
+    @BatchProperty(name = Constants.COS_IS_IBM_CREDENTIAL)
+    String cosCredentialIbm;
+
+    /**
+     * Fhir export format.
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_FHIR_FORMAT)
+    protected String fhirExportFormat;
+
+    /**
+     * Fhir Search from date.
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_FHIR_SEARCH_FROMDATE)
+    String fhirSearchFromDate;
+
+    /**
+     * Fhir search to date.
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_FHIR_SEARCH_TODATE)
+    String fhirSearchToDate;
+
+    /**
+     * Fhir export type filters. Ignored for the basicsystem export
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_FHIR_SEARCH_TYPEFILTERS)
+    String fhirTypeFilters;
+
+    /**
+     * Fhir search page size. Not used
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_FHIR_SEARCH_PAGESIZE)
+    String fhirSearchPageSize;
+
+    @Inject
+    @BatchProperty(name = Constants.INCOMING_URL)
+    String incomingUrl;
+
+    // Control the number of records to read in each "item".
+    @Inject
+    @BatchProperty(name = Constants.COS_BUCKET_FILE_MAX_RESOURCES)
+    String cosBucketFileMaxResources;
+    int resourcesPerItem = Constants.DEFAULT_MAX_RESOURCES_PER_ITEM;
```

Comment:
I think the comment is for resourcesPerItem, not  cosBucketFileMaxResources; no?
```suggestion
    @Inject
    @BatchProperty(name = Constants.COS_BUCKET_FILE_MAX_RESOURCES)
    String cosBucketFileMaxResources;
    
    // Control the number of records to read in each "item".
    int resourcesPerItem = Constants.DEFAULT_MAX_RESOURCES_PER_ITEM;
```

Reply:
Both...one is the string input parameter (optional) and the other is the int value we set from the param. Will make the comment clearer.